updater.go

```diff
@@ -115,10 +116,7 @@ func (u *Updater) update(ctx Context, options UpdateOptions) (*Update, error) {
 	}
 	u.log.Infof("Got update with version: %s", update.Version)
 
-	// Linux updates don't have assets so it's ok to prompt for update above before
-	// we check for nil asset.
-	if update.Asset == nil || update.Asset.URL == "" {
-		u.log.Info("No update asset to apply")
+	if u.updateMissingAsset(update) {
```

Comment:
prefer `update.IsNil()`, callers can log if they want

Reply:
@joshblum update does not have `IsNil()`, is this something you want me to define?

Reply:
yup!